pulp_rpm/app/models/repository.py

```diff
@@ -261,6 +264,63 @@ def on_new_version(self, version):
                 repo_config=self.repo_config,
             )
 
+    @property
+    def all_content_pks(self):
```

Comment:
Is there a reason these all need to be properties as opposed to functions?

Reply:
That one could be a function, but `disk_size` and `on_demand_size` need to be properties as that is how it is defined in pulpcore.

Reply:
Please make this one and `on_demand_artifacts_for_version` into normal functions

Reply:
`on_demand_artifacts_for_version` has to be a static method as that is how it is in pulpcore.

Reply:
Sorry, I misread it as `@property`.  It's fine then